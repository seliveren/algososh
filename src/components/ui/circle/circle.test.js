import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, within} from '@testing-library/react';
import {Circle} from './circle';
import {ElementStates} from "../../../types/element-states";


describe('Test Circle component', () => {
  it("circle without letter is rendered correctly", () => {
    const CircleComponent = renderer.create(<Circle/>).toJSON();
    expect(CircleComponent).toMatchSnapshot();
  });

  it("circle with letter is rendered correctly", () => {
    const props = {
      letter: "5"
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('circle')).toHaveTextContent('5');
    expect(CircleComponent).toMatchSnapshot();
  });

  it("circle with head is rendered correctly", () => {
    const props = {
      head: 'head'
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('head')).toHaveTextContent('head');
    expect(CircleComponent).toMatchSnapshot();
  });

  it("circle with react-element in head is rendered correctly", () => {
    const props = {
      head: <Circle isSmall={true}/>
    };
    render(<Circle {...props} />)
    within(screen.getByTestId('circle')).getByTestId('isSmall circle')
  });

  it("circle with tail is rendered correctly", () => {
    const props = {
      tail: 'tail'
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('tail')).toHaveTextContent('tail');
    expect(CircleComponent).toMatchSnapshot();
  });

  it("circle with react-component in tail is rendered correctly", () => {
    const props = {
      tail: <Circle isSmall={true}/>
    };
    render(<Circle {...props} />)
    within(screen.getByTestId('circle')).getByTestId('isSmall circle')
  });

  it("circle with index is rendered correctly", () => {
    const props = {
      index: '1'
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('index')).toHaveTextContent('1');
    expect(CircleComponent).toMatchSnapshot();
  });

  it("circle with prop isSmall is rendered correctly + default state", () => {
    const props = {
      isSmall: true,
      state: ElementStates.Default
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('isSmall circleColoured')).toBeTruthy();
    expect(screen.getByTestId('isSmall circleColoured')).toHaveClass('default')
    expect(CircleComponent).toMatchSnapshot();
  });

  it("default circle is rendered correctly", () => {
    const props = {
      state: ElementStates.Default
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('circleColoured')).toHaveClass('default')
    expect(CircleComponent).toMatchSnapshot();
  });

  it("changing circle is rendered correctly", () => {
    const props = {
      state: ElementStates.Changing
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('circleColoured')).toHaveClass('changing');
    expect(CircleComponent).toMatchSnapshot();
  });

  it("modified circle is rendered correctly", () => {
    const props = {
      state: ElementStates.Modified
    };
    const CircleComponent = renderer.create(<Circle {...props}/>).toJSON();
    render(<Circle {...props} />);
    expect(screen.getByTestId('circleColoured')).toHaveClass('modified');
    expect(CircleComponent).toMatchSnapshot();
  });

});


