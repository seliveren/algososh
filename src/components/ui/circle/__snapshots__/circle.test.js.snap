// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Circle component changing circle is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component circle with head is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component circle with index is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component circle with letter is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      5
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component circle with prop isSmall is rendered correctly + default state 1`] = `
<div
  className="content "
  data-testid="isSmall circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="isSmall head"
  />
  <div
    className="circle  small default"
    data-testid="isSmall circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component circle with tail is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="tail"
  >
    tail
  </div>
</div>
`;

exports[`Test Circle component circle without letter is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component default circle is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`Test Circle component modified circle is rendered correctly 1`] = `
<div
  className="content "
  data-testid="circle"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="circleColoured"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
